src/hotspot/share/cds/cdsProtectionDomain.cpp

```diff
@@ -47,66 +47,61 @@ OopHandle CDSProtectionDomain::_shared_jar_manifests;
 // the given InstanceKlass.
 // Returns the ProtectionDomain for the InstanceKlass.
 Handle CDSProtectionDomain::init_security_info(Handle class_loader, InstanceKlass* ik, PackageEntry* pkg_entry, TRAPS) {
-  Handle pd;
+  int index = ik->shared_classpath_index();
+  assert(index >= 0, "Sanity");
+  SharedClassPathEntry* ent = FileMapInfo::shared_path(index);
+  Symbol* class_name = ik->name();
 
-  if (ik != NULL) {
```

Comment:
Since the null check for ik has been removed, I'm wondering if it makes sense to add an assert that ik is non-null?
(I noticed the caller already has a null check before calling the function.)

Reply:
There are many functions that require input parameters to be non-NULL without explicitly asserting them. I think in this case, the API is quite obvious that a NULL class shouldn't be passed, so I think an assert is not necessary.